### PR TITLE
add one-import usage for sttp

### PR DESCRIPTION
**Pull request by lihaoyi**
https://github.com/softwaremill/sttp/issues/121

I basically moved the package object into a trait, made the package object inherit from it, and added two `simpleApi` objects (one on JVM, one on JS) that let you `import com.softwaremill.sttp.simpleApi._` and get the normal stuff imported into scope together with a default configuration of the relevant backend (`HttpUrlConnectionBackend` or `FetchBackend`)

Seems to work:
```scala
lihaoyi sttp$ sbt core/test:console

scala> import com.softwaremill.sttp.simpleApi._
import com.softwaremill.sttp.simpleApi._

scala> sttp.get(uri"http://www.google.com").send()
res0: com.softwaremill.sttp.Id[com.softwaremill.sttp.Response[String]] =
Response(Right(<!doctype html><html itemscope="" itemtype="http://schema.org/WebPage" lang="en-SG"><head><meta content="text/html; charset=UTF-8" http-equiv="Content-Type"><meta content="/logos/doodles/2018/gino-bartalis-104th-birthday-5380660112916480-law.gif" itemprop="image"><meta content="Gino Bartali&#8217;s 104th Birthday" property="twitter:title"><meta content="Gino Bartali&#8217;s 104th Birthday #GoogleDoodle" property="twitter:description"><meta content="Gino Bartali&#8217;s 104th Birthday #GoogleDoodle" property="og:description"><meta content="summary_large_image" property="twitter:card"><meta content="@GoogleDoodles" property="twitter:site"><meta content="https://www.google.com/...
```

let's see what CI says

### CODE REVIEW

**Comment by mukel** [comment]
The `FetchBackend` is meant for browsers, it doesn't work on node. Ideally the "default" JS backend should work on both environments similar to `RosHttp`. But that's a story for another PR.

File: core/shared/src/main/scala/com/softwaremill/sttp/SttpApi.scala [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+import java.nio.ByteBuffer
+
+import com.softwaremill.sttp.internal.SttpFile
+
+import scala.annotation.{implicitNotFound, tailrec}
+import scala.collection.immutable.Seq
+import scala.concurrent.duration._
+import scala.language.higherKinds
+
+trait SttpApi extends sttpExtensions {
```
Comment (adamw) [inline core/shared/src/main/scala/com/softwaremill/sttp/SttpApi.scala]: Maybe `sttpExtensions` could now become `SttpApiExtensions`?

File: core/js/src/main/scala/com/softwaremill/sttp/simpleApi.scala [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package com.softwaremill.sttp
+
+object simpleApi extends SttpApi{
```
Comment (adamw) [inline core/js/src/main/scala/com/softwaremill/sttp/simpleApi.scala]: I think I would be leaning towards calling this package `quick`. What do you think?

**Comment by adamw** [comment]
Thanks for the PR! I left two naming notes. Also, I think the README and docs can be updated with the new usage?

**Comment by adamw** [comment]
@mukel if you would know how to write a node backend, that would be great :)

File: core/js/src/main/scala/com/softwaremill/sttp/simpleApi.scala (hunk 2 above)
Comment (lihaoyi) [reply core/js/src/main/scala/com/softwaremill/sttp/simpleApi.scala]: sure

File: core/shared/src/main/scala/com/softwaremill/sttp/SttpApi.scala (hunk 1 above)
Comment (lihaoyi) [reply core/shared/src/main/scala/com/softwaremill/sttp/SttpApi.scala]: Maybe, but this was the existing name so I feel changing it is kind of orthogonal to this PR.

**Comment by lihaoyi** [comment]
I renamed the things to `quick` as you requested

File: core/shared/src/main/scala/com/softwaremill/sttp/SttpApi.scala (hunk 1 above)
Comment (adamw) [reply core/shared/src/main/scala/com/softwaremill/sttp/SttpApi.scala]: That was for consistency, as previously these were extensions to the `sttp` package object. Hence the lower-case name prefix :)

**Comment by lihaoyi** [comment]
capitalized SttpExtensions

**Pull request merged**